packages/api/src/get-export.js

```diff
@@ -0,0 +1,137 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// class to receive POSTS with build information
+
+const firebaseEncode = require('../lib/firebase-encode');
+const { ResourceNotFoundError, handleError } = require('../lib/errors');
+const client = require('./firestore.js');
+
+// 1 = pass, 0 = fail, -1 = not run
+class CsvRow {
+  constructor (buildId, environment, timestamp, totalTestCount, testNameToId) {
+    this.buildId = buildId;
+    this.environment = environment;
+    this.testResults = new Array(totalTestCount).fill(-1);
+    this.testNameToId = testNameToId;
+    this.timestamp = timestamp.toDate();
+  }
+
+  setTestResult (testName, status) {
+    const testIndex = this.testNameToId[testName];
```

Comment:
Is it possible `testName` won't be in `testNameToId`?

Reply:
I think all test names are keys due to the way the dictionary is populated in lines 86-90, but to be safe I added a check here.